operator/internal/probes/broker_test.go

```diff
@@ -0,0 +1,334 @@
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr/testr"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/suite"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/operator/internal/probes"
+	"github.com/redpanda-data/redpanda-operator/operator/internal/testenv"
+	internalclient "github.com/redpanda-data/redpanda-operator/operator/pkg/client"
+	"github.com/redpanda-data/redpanda-operator/pkg/helm"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube"
+	"github.com/redpanda-data/redpanda-operator/pkg/testutil"
+)
+
+func TestIntegrationProber(t *testing.T) {
+	testutil.SkipIfNotIntegration(t)
+
+	suite.Run(t, new(ProberSuite))
+}
+
+type ProberSuite struct {
+	suite.Suite
+
+	manager       ctrl.Manager
+	ctx           context.Context
+	env           *testenv.Env
+	client        client.Client
+	helm          *helm.Client
+	clientFactory *internalclient.Factory
+}
+
+var _ suite.SetupAllSuite = (*ProberSuite)(nil)
+
+func (s *ProberSuite) TestProbes() {
+	chart := s.installChart("default", "")
+	adminClient := s.adminClientFor(chart)
+	defer adminClient.Close()
+
+	// we wrap the first check in a waitFor since there's no guarantee that
+	// the broker will be ready when the helm install completes
+	prober := chart.prober
+	s.waitFor(func(ctx context.Context) (bool, error) {
+		healthy, err := prober.IsClusterBrokerHealthy(s.ctx, 0)
+		if err != nil {
+			s.T().Logf("error checking broker health, retrying: %v", err)
+			return false, nil
+		}
+		return healthy, nil
+	})
+
+	ready, err := prober.IsClusterBrokerReady(s.ctx, 0)
+	s.Require().NoError(err)
+	s.Require().True(ready)
+
+	// now create an under-replicated partition, decommissioning a broker in the process
+	kafkaClient := s.kafkaClientFor(chart)
+	kafkaAdminClient := kadm.NewClient(kafkaClient)
+	defer kafkaAdminClient.Close()
+
+	s.waitFor(func(ctx context.Context) (bool, error) {
+		s.T().Log("attempting to create test-topic")
+		topicResponse, err := kafkaAdminClient.CreateTopic(s.ctx, 1, 3, nil, "test-topic")
+		if err != nil || topicResponse.Err != nil {
+			s.T().Logf("error creating test topic, retrying: %v", errors.Join(err, topicResponse.Err))
+			return false, nil
+		}
+		return true, nil
+	})
+
+	// decommission a broker to make a topic under-replicated
+	err = adminClient.DecommissionBroker(s.ctx, 1)
```

Comment:
Decommission Broker API is asynchronus. You request action, but it would be good to check if Broker finish decommission process.